lib/Dialect/CGGI/IR/CGGIOps.cpp

```diff
@@ -8,17 +8,38 @@
 #include "mlir/include/mlir/IR/ValueRange.h"          // from @llvm-project
 #include "mlir/include/mlir/Support/LLVM.h"           // from @llvm-project
 #include "mlir/include/mlir/Support/LogicalResult.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
 
 namespace mlir {
 namespace heir {
 namespace cggi {
 
 ValueRange Lut2Op::getLookupTableInputs() { return ValueRange{getB(), getA()}; }
 
+LogicalResult Lut2Op::canonicalize(Lut2Op op, PatternRewriter &rewriter) {
+  static const int32_t array2[] = {1, 1};
+  llvm::ArrayRef<int32_t> coeffs2(array2);
+  auto createLutLinCombOp = rewriter.create<LutLinCombOp>(
+      op.getLoc(), op.getOutput().getType(), op.getLookupTableInputs(), coeffs2,
```

Comment:
we needed to use op.getOperands() instead,  and things seem to work out
```bash
./bin/heir-opt   ../tests/cggi/demo.mlir --canonicalize
module {
  func.func @require_post_pass_toposort(%arg0: tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>) -> !lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>> {
    %c0 = arith.constant 0 : index
    %c1 = arith.constant 1 : index
    %c2 = arith.constant 2 : index
    %extracted = tensor.extract %arg0[%c0] : tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>
    %extracted_0 = tensor.extract %arg0[%c1] : tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>
    %extracted_1 = tensor.extract %arg0[%c2] : tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>
    %0 = cggi.lut_lincomb %extracted, %extracted_0, %extracted_1 {coefficients = array<i32: 1, 2, 4>, lookup_table = 8 : ui32} : <encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>
    return %0 : !lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>
  }
}
```

Reply:
thanks for review and notes @asraa

Reply:
Woah, very strange! Thank you for finding that, I'll keep it in mind in case we see it again.